extensions/ql-vscode/test/vscode-tests/no-workspace/data-extensions-editor/model-details/model-details-data-provider.test.ts

```diff
@@ -89,4 +93,45 @@ describe("ModelDetailsDataProvider", () => {
       expect(onDidChangeTreeDataListener).toHaveBeenCalledTimes(1);
     });
   });
+
+  describe("getChildren", () => {
+    const supportedExternalApiUsage = createExternalApiUsage({
+      supported: true,
+    });
+
+    const unsupportedExternalApiUsage = createExternalApiUsage({
+      supported: false,
+    });
+
+    const externalApiUsages: ExternalApiUsage[] = [
+      supportedExternalApiUsage,
+      unsupportedExternalApiUsage,
+    ];
+    const dbItem = mockedObject<DatabaseItem>({
+      getSourceLocationPrefix: () => "test",
+    });
+
+    const usage = createUsage({});
+
+    it("should return [] if item is a usage", async () => {
+      expect(dataProvider.getChildren(usage)).toEqual([]);
+    });
+
+    it("should return usages if item is external api usage", async () => {
+      const externalApiUsage = createExternalApiUsage({ usages: [usage] });
+      expect(dataProvider.getChildren(externalApiUsage)).toEqual([usage]);
+    });
+
+    it("should show all externalApiUsages if hideModeledApis is false and item is undefined", async () => {
```

Comment:
(Minor) The `item is undefined` bit confused me a little bit but I realised what it meant once I got to know the code a bit better. We don't have to change it, but do you think the following description is accurate/better:

```suggestion
    it("should show all externalApiUsages if hideModeledApis is false and looking at the root", async () => {
```

Reply:
sounds good, will rename!